service/history/api/pollupdate/api_test.go

```diff
@@ -0,0 +1,184 @@
+	"go.temporal.io/server/service/history/api/pollupdate"
+	"go.temporal.io/server/service/history/workflow"
+	wcache "go.temporal.io/server/service/history/workflow/cache"
+	"go.temporal.io/server/service/history/workflow/update"
+)
+
+type (
+	mockWFConsistencyChecker struct {
+		api.WorkflowConsistencyChecker
+		GetWorkflowContextFunc func(
+			ctx context.Context,
+			reqClock *clockspb.VectorClock,
+			consistencyPredicate api.MutableStateConsistencyPredicate,
+			workflowKey definition.WorkflowKey,
+			lockPriority workflow.LockPriority,
+		) (api.WorkflowContext, error)
+	}
+
+	mockAPICtx struct {
+		api.WorkflowContext
+		GetUpdateRegistryFunc func(context.Context) update.Registry
+		GetReleaseFnFunc      func() wcache.ReleaseCacheFunc
+	}
+
+	mockReg struct {
+		update.Registry
+		FindFunc func(context.Context, string) (*update.Update, bool)
+	}
+
+	mockUpdateEventStore struct {
+		update.EventStore
+	}
+)
+
+func (mockUpdateEventStore) OnAfterCommit(f func(context.Context))   { f(context.TODO()) }
+func (mockUpdateEventStore) OnAfterRollback(f func(context.Context)) {}
+
+func (m mockWFConsistencyChecker) GetWorkflowContext(
+	ctx context.Context,
+	clock *clockspb.VectorClock,
+	pred api.MutableStateConsistencyPredicate,
+	wfKey definition.WorkflowKey,
+	prio workflow.LockPriority,
+) (api.WorkflowContext, error) {
+	return m.GetWorkflowContextFunc(ctx, clock, pred, wfKey, prio)
+}
+
+func (m mockAPICtx) GetReleaseFn() wcache.ReleaseCacheFunc {
+	return m.GetReleaseFnFunc()
+}
+
+func (m mockAPICtx) GetUpdateRegistry(ctx context.Context) update.Registry {
+	return m.GetUpdateRegistryFunc(ctx)
+}
+
+func (m mockReg) Find(ctx context.Context, updateID string) (*update.Update, bool) {
+	return m.FindFunc(ctx, updateID)
+}
+
+func TestPollOutcome(t *testing.T) {
+	reg := mockReg{}
+	apiCtx := mockAPICtx{
+		GetReleaseFnFunc: func() wcache.ReleaseCacheFunc { return func(error) {} },
+		GetUpdateRegistryFunc: func(context.Context) update.Registry {
+			return reg
+		},
+	}
+	wfcc := mockWFConsistencyChecker{
+		GetWorkflowContextFunc: func(
+			ctx context.Context,
+			reqClock *clockspb.VectorClock,
+			consistencyPredicate api.MutableStateConsistencyPredicate,
+			workflowKey definition.WorkflowKey,
+			lockPriority workflow.LockPriority,
+		) (api.WorkflowContext, error) {
+			return apiCtx, nil
+		},
+	}
+
+	updateID := t.Name() + "-update-id"
+	req := historyservice.PollWorkflowExecutionUpdateRequest{
+		Request: &workflowservice.PollWorkflowExecutionUpdateRequest{
+			UpdateRef: &updatepb.UpdateRef{
+				WorkflowExecution: &commonpb.WorkflowExecution{
+					WorkflowId: t.Name() + "-workflow-id",
+					RunId:      t.Name() + "-run-id",
+				},
+				UpdateId: updateID,
+			},
+		},
+	}
+
+	t.Run("update not found", func(t *testing.T) {
+		reg.FindFunc = func(ctx context.Context, updateID string) (*update.Update, bool) {
+			return nil, false
+		}
+		_, err := pollupdate.Invoke(context.TODO(), &req, wfcc)
```

Comment:
I use TODO because I'm moderatly serious about keeping things to 80 chars. I can restructure to use Background with a separate var declaration.

Reply:
Oh but FYI I did it in the other PR

Reply:
I think Background is more appropriate. We'll probably want to do some context propagation refactor later and search for all context.TODOs, so this would save us some time later.